.github/workflows/haskell.yml

```diff
@@ -65,8 +66,17 @@ jobs:
           ${{ runner.os }}-${{ matrix.ghc }}-${{ env.cache-name }}-
 
     - name: Install cabal dependencies
+      id: build-dependencies
       run: cabal build --only-dependencies --enable-tests --enable-benchmarks all
 
+    - name: "Save cache"
+      uses: actions/cache/save@v4
+      id: save-cabal-cache
+      if: steps.build-dependencies.outcome == 'success' && steps.restore-cabal-cache.outputs.cache-hit != 'true'
```

Comment:
you want to save cache also when stuff succeeded, because when we bump `index-state` the cache will be outdated.

or is `cache-hit` `'true'` only when the primary key is hit? If so, add a comment.

Reply:
> Note: cache-hit will be set to true only when cache hit occurs for the exact key match. For a partial key match via restore-keys or a cache miss, it will be set to false.

I'll add a comment